File: android/src/main/java/com/mattermost/networkclient/enums/WebSocketEvents.kt
package com.mattermost.networkclient.enums

enum class WebSocketEvents(val event: String) {
    OPEN_EVENT("WebSocketClient-Open"),
    CLOSING_EVENT("WebSocketClient-Closing"),
    CLOSE_EVENT("WebSocketClient-Close"),
    ERROR_EVENT("WebSocketClient-Error"),
    MESSAGE_EVENT("WebSocketClient-Message"),
    READY_STATE_EVENT("WebSocketClient-ReadyState")
}

File: ios/WebSocket/WebSocketEvents.swift
import Foundation

@objc public protocol WebSocketClientDelegate {
    func sendEvent(name: String, result: Dictionary<String, Any>?)
}

enum WSEvents: String, CaseIterable {
    case OPEN_EVENT = "WebSocketClient-Open"
    case CLOSE_EVENT = "WebSocketClient-Close"
    case ERROR_EVENT, CLIENT_ERROR = "WebSocketClient-Error"
    case MESSAGE_EVENT = "WebSocketClient-Message"
    case READY_STATE_EVENT = "WebSocketClient-ReadyState"
}

extension WebSocketWrapper {
    @objc
    public static var supportedEvents: [String] {
        return WSEvents.allCases.map(\.rawValue)
    }
}
